File: scripts/track-skill-invocation.py
#!/usr/bin/env python3
"""
Track Skill Invocations - Records when a skill is invoked.
Hook: PostToolUse (tool_name == "Skill")

Works with semantic-skill-router.py to provide routing feedback.
"""

import json
import sys
from pathlib import Path

TRACKING_DIR = Path.home() / ".claude" / "routing-tracking"

def main():
    # Read input from stdin (JSON from Claude Code hook)
    try:
        input_data = json.load(sys.stdin)
    except json.JSONDecodeError:
        sys.exit(0)

    tool_name = input_data.get("tool_name", "")

    # Only track Skill invocations
    if tool_name != "Skill":
        sys.exit(0)

    # Extract skill name from tool input
    tool_input = input_data.get("tool_input", {})
    skill_name = tool_input.get("skill", "")

    if not skill_name:
        sys.exit(0)

    # Save invocation
    TRACKING_DIR.mkdir(parents=True, exist_ok=True)
    invocation_file = TRACKING_DIR / "last-invocation.json"

    data = {
        "skill_name": skill_name,
        "timestamp": __import__("time").time()
    }

    invocation_file.write_text(json.dumps(data), encoding="utf-8")

if __name__ == "__main__":
    main()

File: scripts/track-skill-invocation.js
#!/usr/bin/env node
/**
 * Track Skill Invocations - Records when a skill is invoked.
 * Hook: PostToolUse (tool_name == "Skill")
 *
 * Works with semantic-skill-router to provide routing feedback.
 */

const fs = require('fs');
const path = require('path');
const os = require('os');

// Add lib directory to path for imports
const libDir = path.join(__dirname, 'lib');
const debugLogger = require(path.join(libDir, 'debug-logger.js'));

const { logHookStart, logHookEnd, logDebug } = debugLogger;

const TRACKING_DIR = path.join(os.homedir(), '.claude', 'routing-tracking');

function handleInvocation(data) {
    logDebug('PostToolUse', 'track-skill-invocation.js', 'Input parsed successfully', 'INFO');

    const toolName = data.tool_name || '';

    // Only track Skill invocations
    if (toolName !== 'Skill') {
        logDebug('PostToolUse', 'track-skill-invocation.js', `Not a Skill invocation: ${toolName}`, 'SKIP');
        return false;
    }

    const toolInput = data.tool_input || {};
    const skillName = toolInput.skill || '';

    if (!skillName) {
        logDebug('PostToolUse', 'track-skill-invocation.js', 'No skill name found', 'SKIP');
        return false;
    }

    logDebug('PostToolUse', 'track-skill-invocation.js', `Tracking skill: ${skillName}`, 'INFO');

    // Save invocation
    if (!fs.existsSync(TRACKING_DIR)) {
        fs.mkdirSync(TRACKING_DIR, { recursive: true });
    }

    const invocationFile = path.join(TRACKING_DIR, 'last-invocation.json');
    const trackingData = {
        skill_name: skillName,
        timestamp: Date.now() / 1000
    };

    fs.writeFileSync(invocationFile, JSON.stringify(trackingData), 'utf-8');

    logDebug('PostToolUse', 'track-skill-invocation.js', 'Invocation saved', 'INFO');
    return true;
}

/* istanbul ignore next */
function main() {
    logHookStart('PostToolUse', 'track-skill-invocation.js');

    let inputData = '';
    process.stdin.setEncoding('utf8');

    process.stdin.on('readable', () => {
        let chunk;
        while ((chunk = process.stdin.read()) !== null) {
            inputData += chunk;
        }
    });

    process.stdin.on('end', () => {
        try {
            const data = JSON.parse(inputData);
            const success = handleInvocation(data);
            logHookEnd('PostToolUse', 'track-skill-invocation.js', success);
            process.exit(0);
        } catch (e) {
            logDebug('PostToolUse', 'track-skill-invocation.js', `ERROR: ${e.message}`, 'ERROR');
            logHookEnd('PostToolUse', 'track-skill-invocation.js', false);
            process.exit(0);
        }
    });
}

// Export for testing
/* istanbul ignore else */
if (typeof module !== 'undefined' && module.exports) {
    module.exports = { handleInvocation, TRACKING_DIR };
}

// Run main if executed directly
/* istanbul ignore if */
if (require.main === module) {
    main();
}
